server/controllers/v1/post.ts

```diff
@@ -105,6 +106,74 @@ export default class PostController {
     };
   }
 
+  async getPostsByQuery(
+    userID: string,
+    query: string,
+    limit: number,
+    offset: number
+  ): Promise<{
+    status: number;
+    data: { result?: PostUserPreview[]; count: number; total: number };
+  }> {
+    await db.sequelize.query(`DROP TABLE IF EXISTS "PostSearches";`);
+    await db.sequelize.query(
+      `CREATE TEMP TABLE "PostSearches" AS 
+      SELECT *, (
+        setweight(to_tsvector(coalesce("title",'')), 'A') || 
+        setweight(to_tsvector(coalesce("location",'')), 'C') || 
+        setweight(to_tsvector(coalesce("body",'')), 'D')
+      ) as "tsvector" 
+      FROM "Posts";`
+    );
```

Comment:
We don't want to do this on each and every request. 

I'm assuming that the PostSearches table is used to cache the search query. This means we have to build the table per query but what if we have lots of data? How long will that take? 

Second, what if I make a query, the `PostSearches` table is created and built, then before the `select` is executed below for me, you make a search and `DROP TABLE IF EXISTS "PostSearches"` is run and then my `select` is run?